pkg/state/state_skel.go

```diff
@@ -135,6 +134,44 @@ func (s *stateSkel) createOrUpdateObjs(
 	return nil
 }
 
+func (s *stateSkel) mergeObjects(updated, current *unstructured.Unstructured) error {
+	// Set resource version
+	// ResourceVersion must be passed unmodified back to the server.
+	// ResourceVersion helps the kubernetes API server to implement optimistic concurrency for PUT operations
+	// when two PUT requests are specifying the resourceVersion, one of the PUTs will fail.
+	updated.SetResourceVersion(current.GetResourceVersion())
+
+	gvk := updated.GroupVersionKind()
+	if gvk.Group == "" && gvk.Kind == "ServiceAccount" {
+		return s.mergeServiceAccount(updated, current)
+	}
+	return nil
+}
+
+// For Service Account, keep secrets if exists
+func (s *stateSkel) mergeServiceAccount(updated, current *unstructured.Unstructured) error {
+	curSecrets, ok, err := unstructured.NestedSlice(current.Object, "secrets")
```

Comment:
nit: we could save some code dupl if a `mergeNestedSlice(updated, current, sliceFieldName)` is defined.

not critical at this point, but if we end up adding more, this would need to be refactored.